### PR TITLE
Fix javadoc package groups and exclude implementation packages

### PR DESCRIPTION
**Pull request by srnagar**
Some packages in application configuration were grouped under "Other packages" in javadocs. Also, javadocs were generated for `*.implementation` packages. 

This change fixes both these issues for track 2 modules.

### CODE REVIEW

File: pom.client.xml [hunk 1]
```diff
@@ -260,14 +260,14 @@
...
-              *.implementation.*:
+              *.impl*:
+              *.implementation*:
               com.azure.tracing.opencensus*
             </excludePackageNames>
             <groups>
               <group>
                 <title>Azure Core</title>
-                <packages>com.azure.core:com.azure.core.annotations:com.azure.core.credentials:com.azure.core.exception:com.azure.core.http*</packages>
+                <packages>com.azure.core:com.azure.core.annotations:com.azure.core.credentials:com.azure.core.exception:com.azure.core.http*:com.azure.core.configuration:com.azure.core.util</packages>
```
Comment (JonathanGiles) [inline pom.client.xml]: Can you also support all packages under the util package please?

Comment (srnagar) [reply pom.client.xml]: There's nothing else under `util`. Do you still want me to add `util*` just in case we add something later?

Comment (JonathanGiles) [reply pom.client.xml]: Yes, I just proposed moving some packages under util, so doing that eagerly now saves doing it later.

Comment (srnagar) [reply pom.client.xml]: Ok sure, will update it.

Comment (srnagar) [reply pom.client.xml]: Done

**Pull request merged**